File: app/src/main/java/io/vinter/trackmyanime/entity/top/Top.java
package io.vinter.trackmyanime.entity.top;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.vinter.trackmyanime.entity.top.AnimeTop;

public class Top {

    @SerializedName("request_hash")
    @Expose
    private String requestHash;
    @SerializedName("request_cached")
    @Expose
    private Boolean requestCached;
    @SerializedName("request_cache_expiry")
    @Expose
    private Integer requestCacheExpiry;
    @SerializedName("top")
    @Expose
    private List<AnimeTop> top = null;

    public String getRequestHash() {
        return requestHash;
    }

    public void setRequestHash(String requestHash) {
        this.requestHash = requestHash;
    }

    public Boolean getRequestCached() {
        return requestCached;
    }

    public void setRequestCached(Boolean requestCached) {
        this.requestCached = requestCached;
    }

    public Integer getRequestCacheExpiry() {
        return requestCacheExpiry;
    }

    public void setRequestCacheExpiry(Integer requestCacheExpiry) {
        this.requestCacheExpiry = requestCacheExpiry;
    }

    public List<AnimeTop> getTop() {
        return top;
    }

    public void setTop(List<AnimeTop> top) {
        this.top = top;
    }

}

File: app/src/main/java/io/vinter/trackmyanime/entity/top/Top.kt
package io.vinter.trackmyanime.entity.top

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import io.vinter.trackmyanime.entity.top.AnimeTop

class Top {

    @SerializedName("request_hash")
    @Expose
    var requestHash: String? = null
    @SerializedName("request_cached")
    @Expose
    var requestCached: Boolean? = null
    @SerializedName("request_cache_expiry")
    @Expose
    var requestCacheExpiry: Int? = null
    @SerializedName("top")
    @Expose
    var top: List<AnimeTop>? = null

}
